webapp/sources/rudder/rudder-core/src/main/scala/com/normation/cfclerk/domain/Constraint.scala

```diff
@@ -157,12 +173,15 @@ object MailVType extends FixedRegexVType {
 final case class IntegerVType(regex: Option[RegexConstraint] = None) extends VTypeConstraint with VTypeWithRegex  {
   override val name = "integer"
   override def getFormatedValidated(value:String, forField:String, escapeString: String => String) : PureResult[String] = {
-    super.getFormatedValidated(value, forField, escapeString).flatMap( _ =>
-      (try {
-        Right(value.toInt)
-      } catch {
-        case _:Exception => Left(Inconsistency(s"Wrong value ${value} for field '${forField}': expecting an integer."))
-      }).map( _ => value )
+    super.getFormatedValidated(value, forField, escapeString).flatMap( escaped =>
+      VTypeConstraint.checkIsProperty(value, forField) { v =>
+        try {
+          v.toInt
+          Right(())
```

Comment:
why is it returning an empty Right ?

Reply:
to force the user (me) to `map(_ => escaped)` for the return value. We don't use these value, the parsing/etc is just done to check if the provided input is of the correct type, but we need to use the escaped string char that is already computed before that method is called